python/pyarrow/array.pxi

```diff
@@ -704,6 +707,95 @@ def _restore_array(data):
     return pyarrow_wrap_array(MakeArray(ad))
 
 
+cdef class ArrayStatistics(_Weakrefable):
+    """
+    The class for statistics of an array.
+    """
+
+    def __init__(self):
+        raise TypeError("Do not call {}'s constructor directly"
+                        .format(self.__class__.__name__))
+
+    cdef void init(self, const shared_ptr[CArrayStatistics]& sp_statistics) except *:
```

Comment:
`except *` means it could raise Python exceptions, but it doesn't here, so perhaps you can remove that annotation (though it's not really a problem either).

Reply:
Thanks! I didn't know much about `except` in Cython...